### PR TITLE
Update web platform tests and add a bunch of HTML element stubs

### PR DESCRIPTION
**Pull request by domenic**
Blocked on https://github.com/w3c/web-platform-tests/pull/2107

### CODE REVIEW

**Comment by Sebmaster** [comment]
Need to update web-platform-tests again to get the upstream fixes.

File: test/w3c/w3ctest.js [hunk 1]
```diff
@@ -1,30 +1,32 @@
...
-var toFileUrl = require("../../lib/jsdom/utils").toFileUrl;
+const http = require("http");
+const ecstatic = require("ecstatic");
+const path = require("path");
+const jsdom = require("../..");
 
-function createJsdom(source, url, t) {
-  const reporterHref = toFileUrl(__dirname + "/tests/resources/testharnessreport.js");
+const hostname = "127.0.0.1";
+const port = "9999";
```
Comment (Sebmaster) [inline test/w3c/w3ctest.js]: I'd use a dynamic port instead since that ensures the port is free.

File: test/w3c/w3ctest.js [hunk 2]
```diff
@@ -50,40 +52,21 @@ function createJsdom(source, url, t) {
...
 
-    loaded: function (err) {
-      t.ifError(err && err[0].data.error);
+    loaded(err) {
+      t.ifError(err);
     }
   });
 }
 
-function testUrl(url, t) {
```
Comment (Sebmaster) [inline test/w3c/w3ctest.js]: We don't allow uncloned w3c tests anymore? :(

Comment (domenic) [inline test/w3c/w3ctest.js]: That was the idea... but I can add it back I guess.

**Comment by domenic** [comment]
Merged without the server commit and with the most recent updates.

**Pull request merged**